### PR TITLE
Adding LSM module (x/lsnative) to the app

### PR DESCRIPTION
**Pull request by xlab**
## 1. Overview

This PR includes an LSM module from persistence-sdk into the app. Nencessary package path replacements made.

## 2. Implementation details

It uses a special version `github.com/persistenceOne/persistence-sdk/v2 v2.0.1-0.20230403194140-3773b08f31a1`
that corresponds to https://github.com/persistenceOne/persistence-sdk/pull/391 PR in persistence-sdk repo.

## 3. How to test/use

```
$ git clone git@github.com:persistenceOne/persistenceCore.git
$ git checkout max/add-lsnative
$ make install
$ cd scripts/
$ ./setup.sh
$ ./startup.sh
```

In a separate tab, while chain is running:
```
$ persistenceCore --keyring-backend=test tx --from=test --gas=auto --gas-adjustment=1.5 staking tokenize-share persistencevaloper1pkkayn066msg6kn33wnl5srhdt3tnu2v94kvz9 10000000stake persistence1pkkayn066msg6kn33wnl5srhdt3tnu2vv3k3tu --yes

$ persistenceCore q bank balances persistence1pkkayn066msg6kn33wnl5srhdt3tnu2vv3k3tu
balances:
- amount: "10000000"
  denom: persistencevaloper1pkkayn066msg6kn33wnl5srhdt3tnu2v94kvz9/1
- amount: "100000068117444"
  denom: stake
pagination:
  next_key: null
  total: "0"
```

## 4. Future Work (optional)

Adding an automatic e2e test that runs the app and uses an LSM scenario.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -413,7 +415,7 @@ func (app *Application) ExportAppStateAndValidators(forZeroHeight bool, jailWhit
 
 		for ; kvStoreReversePrefixIterator.Valid(); kvStoreReversePrefixIterator.Next() {
 			addr := sdk.ValAddress(kvStoreReversePrefixIterator.Key()[1:])
-			validator, found := app.StakingKeeper.GetValidator(context, addr)
+			validator, found := app.StakingKeeper.GetLiquidValidator(context, addr)
```
Comment (puneet2019) [inline app/app.go:418]: why only getLiquidValidator?

Comment (xlab) [reply app/app.go:418]: GetLiquidValidator is an in-place replacement for GetValidator, it uses upgraded `` that allows to read extra fields that are related to ls :)

See the implementation difference between two:
https://github.com/persistenceOne/persistence-sdk/blob/3773b08f31a14f04c6f804851449b55979997bab/x/lsnative/staking/keeper/validator.go#L16-L39
(only `MustUnmarshalValidator`)

Comment (xlab) [reply app/app.go:418]: type `Validator` from the forked version is extended with these fields:

Comment (xlab) [reply app/app.go:418]: type `Validator` from the forked version is extended with these fields:
Screenshot 2023-04-04 at 12 48 20

**Review by ajeet97 — APPROVED** [review]

File: app/app.go (hunk 1 above)
Comment (puneet2019) [reply app/app.go:418]: got it thanks!

**Review by puneet2019 — APPROVED** [review]

**Pull request merged**